app/main.py

```diff
@@ -1,34 +1,77 @@
+from dataclasses import dataclass
+
+
+@dataclass
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    row: int
+    column: int
+    is_alive: bool = True
+
+    def hit(self) -> None:
+        self.is_alive = False
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple[int, int], end: tuple[int, int], is_drowned=False) -> None:
```

Comment:
Consider adding validation to ensure that the start and end coordinates form a valid ship (either horizontal or vertical). Currently, the code assumes that the input is always correct, which might not be the case.